src/diagnostic.rs

```diff
@@ -1,25 +1,23 @@
 // Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
 use serde::Serialize;
-use std::convert::TryInto;
 
 #[derive(Debug, Clone, Copy, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Position {
-  pub line: usize,
-  pub col: usize,
+  pub line_index: usize,
+  pub column_index: usize,
```

Comment:
This should all be 0-index now except when displayed or in the tests. (Still that weird 1-indexed line and 0-indexed column in the tests... maybe we could change that later to both 1-indexed over there)

Reply:
Keep in mind that this is used by LSP/vscode_deno, so changing it might break the former

Reply:
I think we shouldn't use `usize` everywhere to represent line/column numbers because overuse of `usize` may result in developers' confusion in whether this type expresses 0-indexed value that is internally used or 1-indexed value that is for displaying diagnostics to end users. Instead, we could create new types, one representing internal values and the other values to be displayed.

---

Oh I found https://github.com/dsherret/text_lines which provides new types I described above. Maybe we could utilize it later on.

Reply:
I'll add some comments to indicate it's 0-indexed. I think using `usize` is useful here because then you can index these types into an array of lines or a characters.

Yup, we could use text_lines in the future.

Reply:
I see, that makes sense!